subsys/nrf_security/src/CMakeLists.txt

```diff
@@ -203,7 +203,7 @@ endif()
 add_subdirectory(drivers)
 
 # Add legacy Mbed TLS APIs
-if(CONFIG_MBEDTLS_LEGACY_CRYPTO_C)
+if(CONFIG_MBEDTLS_LEGACY_CRYPTO_C OR (CONFIG_NRF_OBERON AND CONFIG_BUILD_WITH_TFM))
```

Comment:
Why `AND CONFIG_BUILD_WITH_TFM`?

Reply:
Because we use the same code in both TF-M (secure image) and NS image builds. We don't want to load legacy libraries just based on `CONFIG_NRF_OBERON`. Checking that `CONFIG_BUILD_WITH_TFM` is set means we are in a NS image